napari/_qt/widgets/_tests/test_qt_dims.py

```diff
@@ -173,7 +173,7 @@ def test_singleton_dims(qtbot):
     """
     ndim = 4
     dims = Dims(ndim=ndim)
-    dims.set_range(0, (0, 1, 1))
+    dims.set_range(0, (0, 0, 1))
```

Comment:
This probably changed in the previous thick slicing PR, but it would be good to clarify (i.e. in the `Dims` docstring) that `Dims.range` uses a closed interval / inclusive upper bound unlike Python's built-in `range` and numpy's `arange`.

Reply:
Sure. I would say that I don't think it's helpful to compare this to `range` and `arange`, and we should rather compare it to `linspace`. In fact, it might be more intuitive (for the future) to have `nsteps` as the third number rather than `step`.

Reply:
I'm going to try my luck here again and actually broaden the scope of my ask. I think we should ensure the docstring of at least every public attribute/property where it matters (e.g. extent, extent_augmented, corner_pixels, etc.) describes if it uses an inclusive/exclusive lower/upper bound.

After this PR I think these things are mostly closed intervals, but since we're changing the behavior for some of them, it could be really useful to clarify the new behavior in the docstrings to remove any confusion users and developers.

Reply:
Ok, I added the "incluse bounds" info on all public places where it's relevant.